src/capi.rs

```diff
@@ -356,6 +379,42 @@ pub unsafe extern fn rav1e_config_default() -> *mut Config {
   Box::into_raw(c)
 }
 
+/// Setup a single pass rate control
+///
+#[no_mangle]
+pub unsafe extern fn rav1e_config_set_rc_single_pass(cfg: *mut Config) {
+  (*cfg).cfg.rate_control.summary = None;
+}
+
+/// Setup a second pass rate control using the provided summary
+///
+#[no_mangle]
+pub unsafe extern fn rav1e_config_set_rc_summary(
+  cfg: *mut Config, buf: *mut u8, len: size_t,
+) -> c_int {
+  let bytes = slice::from_raw_parts(buf, len);
+  let summary = rav1e::RateControlSummary::from_slice(bytes).ok();
+
+  if summary.is_none() {
+    -1
+  } else {
+    (*cfg).cfg.rate_control.summary = summary;
+
+    0
+  }
+}
+
+/// Request to emit pass data
```

Comment:
Related to the comment on the `rav1e_config_set_rc_single_pass`, the requirements and restrictions on this fuction *must* be documented. How to call functions and in what order, and what depends on what?

Reply:
It doesn't immediately trigger a validation so you can call it anytime, same as `rav1e_config_set_rc_single_pass`

Reply:
I do not mean just validation. There is no documentation provided at all on how to use this API correctly currently. A user most figure out the (IMO very non-straightforward) API calls, in what order, and which to call, to us the API. I am not a proponent of this.

Reply:
We can add more explanations in general.